catt/controllers.py

```diff
@@ -215,32 +219,29 @@ def __init__(self):
 
         if not self.store_path.is_file():
             devices = get_chromecasts()
-            cache_data = {d.name: self._create_device_entry(d.host, d.port) for d in devices}
+            cache_data = {
+                d.name: CCInfo(d.host, d.port, d.device.manufacturer, d.model_name, d.cast_type).all_info
+                for d in devices
+            }
             self._write_store(cache_data)
 
-    def _create_device_entry(self, ip, port):
-        device_data = {"ip": ip}
-        if port != DEFAULT_PORT:
-            device_data["group_port"] = port
-        return device_data
-
     def get_data(self, name: str):  # type: ignore
         data = self._read_store()
         # In the case that cache has been initialized with no cc's on the
         # network, we need to ensure auto-discovery.
         if not data:
-            return (None, None)
+            return None
         if name:
             fetched = data.get(name)
         else:
             # When the user does not specify a device, we need to make an attempt
             # to consistently return the same IP, thus the alphabetical sorting.
             fetched = data[min(data, key=str)]
-        return (fetched["ip"], fetched.get("group_port", 0)) if fetched else (None, None)
+        return CCInfo(**fetched) if fetched else None
 
-    def set_data(self, name: str, ip: str, port: int) -> None:  # type: ignore
+    def set_data(self, name: str, device_entry: CCInfo) -> None:  # type: ignore
```

Comment:
What's the error here that made us ignore the type?

Reply:
Removing the ignore results in this error msg from `mypy`:
```
Signature of "set_data" incompatible with supertype "CattStore"
```
..which still doesn't make sense to me.